apis/python/src/tiledbsoma/io/ingest.py

```diff
@@ -1170,24 +1158,43 @@ def is_str_cat_col(field: pa.Field) -> bool:
             if name not in old_schema.names:
                 continue
             column = arrow_table.column(name)
-            old_info = old_schema.field(name)
-            new_info = new_schema.field(name)
-            if is_str_col(old_info) and is_str_cat_col(new_info):
-                # Convert from categorical-of-string to plain string.
-                column = column.to_pylist()
-            elif is_str_cat_col(old_info) and is_str_col(new_info):
-                # Convert from plain string to categorical-of-string.  Note:
+            old_field = old_schema.field(name)
+            new_field = new_schema.field(name)
+            if not is_cat(old_field) and is_cat(new_field):
+                # Convert from categorical to non-categorical.  Note that if
+                # this is a pa.ChunkedArray, there is no .dictionary_decode()
+                # for it, but each chunk does have a .dictionary_decode().
+
+                if isinstance(column, pa.ChunkedArray):
+                    column = pa.chunked_array(
+                        [chunk.dictionary_decode() for chunk in column.chunks]
+                    )
+                elif isinstance(column, pa.DictionaryArray):
+                    column = column.dictionary_decode()
+                else:
+                    column = column.to_pylist()
```

Comment:
I'm still confused why this branch needs to round-trip through a Python list.  As I understand the code, you should be able to just leave it as-is, i.e., as an arrow Array.   `Tables.from_pydict` will happily consume that....

Reply:
Actually, thinking about this more -- under what conditions will this branch ever get used?   The column is a dict type, and the first two branches cover chunked and not chunked.  Can this ever execute?

If not, it should be an assert/failure, right, as it indicates that something unexpected made it past line 1163